test/breaker_box_test.exs

```diff
@@ -10,231 +10,246 @@ defmodule BreakerBoxTest do
 
-      assert {:ok, @non_existent_breaker} = BreakerBox.status(@non_existent_breaker)
+      assert {:ok, @non_existent_breaker} =
+               BreakerBox.status(@non_existent_breaker, @process_name)
 
       expected_config = @test_breaker_config
 
-      assert {:ok, ^expected_config} = BreakerBox.get_config(@non_existent_breaker)
+      assert {:ok, ^expected_config} = BreakerBox.get_config(@non_existent_breaker, @process_name)
 
-      assert :ok = BreakerBox.remove(@non_existent_breaker)
+      assert :ok = BreakerBox.remove(@non_existent_breaker, @process_name)
 
       assert {:error, {:breaker_not_found, @non_existent_breaker}} =
-               BreakerBox.status(@non_existent_breaker)
+               BreakerBox.status(@non_existent_breaker, @process_name)
     end
 
     test "re-registering a breaker will override the original configuration" do
       assert {:error, {:breaker_not_found, @non_existent_breaker}} =
-               BreakerBox.status(@non_existent_breaker)
+               BreakerBox.status(@non_existent_breaker, @process_name)
 
-      assert :ok = BreakerBox.register(@non_existent_breaker, @test_breaker_config)
+      assert :ok = BreakerBox.register(@non_existent_breaker, @test_breaker_config, @process_name)
 
-      assert {:ok, @non_existent_breaker} = BreakerBox.status(@non_existent_breaker)
+      assert {:ok, @non_existent_breaker} =
+               BreakerBox.status(@non_existent_breaker, @process_name)
 
       expected_config = @test_breaker_config
       new_config = BreakerConfiguration.reset_after_milliseconds(@test_breaker_config, 999_999)
 
-      assert {:ok, ^expected_config} = BreakerBox.get_config(@non_existent_breaker)
+      assert {:ok, ^expected_config} = BreakerBox.get_config(@non_existent_breaker, @process_name)
 
-      assert :ok = BreakerBox.register(@non_existent_breaker, new_config)
+      assert :ok = BreakerBox.register(@non_existent_breaker, new_config, @process_name)
 
-      assert {:ok, @non_existent_breaker} = BreakerBox.status(@non_existent_breaker)
+      assert {:ok, @non_existent_breaker} =
+               BreakerBox.status(@non_existent_breaker, @process_name)
 
-      assert {:ok, ^new_config} = BreakerBox.get_config(@non_existent_breaker)
+      assert {:ok, ^new_config} = BreakerBox.get_config(@non_existent_breaker, @process_name)
 
-      assert :ok = BreakerBox.remove(@non_existent_breaker)
+      assert :ok = BreakerBox.remove(@non_existent_breaker, @process_name)
 
       assert {:error, {:breaker_not_found, @non_existent_breaker}} =
-               BreakerBox.status(@non_existent_breaker)
+               BreakerBox.status(@non_existent_breaker, @process_name)
+    end
+
+    test "registering a breaker with max_failures less then 5 returns error" do
+      invalid_breaker_config = Map.put(@test_breaker_config, :max_failures, 2)
+
+      assert {:error, "BreakerBox: max_failures must be at least 5"} =
```

Comment:
Again, less than 5 is perfectly fine, just has to be more than 1 at the moment, unless we figure out a user-friendly way to handle that.